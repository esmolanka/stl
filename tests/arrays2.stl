
type Table =
  exists (n : Nat).
  { name : String[n]
  , a    : Int[n]
  , b    : Int[n]
  }

type SumProduct =
  (exists (n : Nat).
     { a : Int[n]
     , b : Int[n]
     }) -> Int

#check SumProduct <: Table -> Int
#check Table -> Int <: SumProduct
