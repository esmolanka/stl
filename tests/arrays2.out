tests/arrays2.stl:15:1-15:34:
  Subtype:
    (∃ (n : Nat). (∃ (ρ : Row). {a : Int[n], b : Int[n] | ρ})) -> Int
  Supertype:
    (∃ (n : Nat). (∃ (ρ : Row). {name : String[n], a : Int[n], b : Int[n] | ρ})) -> Int
  Result:
    OK
tests/arrays2.stl:16:1-16:34:
  Subtype:
    (∃ (n : Nat). (∃ (ρ : Row). {name : String[n], a : Int[n], b : Int[n] | ρ})) -> Int
  Supertype:
    (∃ (n : Nat). (∃ (ρ : Row). {a : Int[n], b : Int[n] | ρ})) -> Int
  Result:
    FAIL
    Cannot unify types.
        ▫︎
      Is not subsumed by:
        ▪︎